tests/draft2019-09/optional/format/uuid.json

```diff
@@ -0,0 +1,45 @@
+[
+    {
+        "description": "uuid format",
+        "schema": {
+            "format": "uuid"
+        },
+        "tests": [
+            {
+                "description": "all upper-case",
+                "data": "2EB8AA08-AA98-11EA-B4AA-73B441D16380",
+                "valid": true
+            },
+            {
+                "description": "all lower-case",
+                "data": "2eb8aa08-aa98-11ea-b4aa-73b441d16380",
+                "valid": true
+            },
+            {
+                "description": "mixed case",
+                "data": "2eb8aa08-AA98-11ea-B4Aa-73B441D16380",
+                "valid": true
+            },
+            {
+                "description": "all zeroes is valid",
+                "data": "00000000-0000-0000-0000-000000000000",
+                "valid": true
+            },
+            {
+                "description": "wrong length",
+                "data": "2eb8aa08-aa98-11ea-b4aa-73b441d1638",
+                "valid": false
+            },
+            {
+                "description": "missing section",
+                "data": "2eb8aa08-aa98-11ea-73b441d16380",
+                "valid": false
+            },
+            {
+                "description": "bad characters",
+                "data": "2eb8aa08-aa98-11ea-b4ga-73b441d16380",
+                "valid": false
+            }
+        ]
+    }
```

Comment:
Also also probably good to have a UUID *6* (i.e. by inserting 6 where the version goes) and assert it's invalid?

Reply:
good call. I've seen some regexes that are tighter and only accept the current versions, but indeed the RFC says

> Apart from determining whether the timestamp portion of the UUID is in the future and therefore not yet assignable, there is no mechanism for determining whether a UUID is 'valid'.

..so these implementations are being needlessly (and invalidly) too strict.

Reply:
*in*valid, or valid? IMO we should accept future versions as well.

Reply:
fwiw I just tried a "future version" uuid in PostgreSQL against its uuid data type and it accepted version 6 and version 15 (f).

Reply:
Cool, yep good catch (I didn't go read the RFC first, so yeah seems valid is the right behavior)!